src/coordinator/coordinator.rs

```diff
@@ -408,7 +407,16 @@ impl Coordinator {
     }
 
     fn check_merge_type(&self, elevator_data: ElevatorData) -> MergeType {
-        let new_elevators = elevator_data.states.keys().len() > self.elevator_data.states.keys().len();
+        let mut new_elevators = false;
+        for key in self.elevator_data.states.keys() {
```

Comment:
Is there a case for when an elevator isn't in the list? Should the merges be triggered whenever the states ID's differ? And not just if one contains more ID's than another? I'm not sure.